### PR TITLE
v4 - testing needed

### PR DESCRIPTION
**Pull request by asheroto**
Hey fellow contributors!

@1ckov, @MisterZeus, @ChrisTitusTech, @uffemcev

Microsoft published some short URLs and an easy method to install winget. Because of that, I was able to remove a **huge chunk** of the `winget-install` code.

<details><summary>Here's what they now have on the winget page.</summary>

<p>

https://learn.microsoft.com/en-us/windows/package-manager/winget/#install-winget-on-windows-sandbox

image

</p>
</details> 

I have tested the code for `winget-install` v4 on the following operating systems:

- Windows 10
- Windows 11
- Server 2022
- Windows Sandbox

All seem to work great.

Can you guys test on your end and tell me if this new code is working for you too?

```powershell
&([ScriptBlock]::Create((irm https://raw.githubusercontent.com/asheroto/winget-install/v4/winget-install.ps1))) -Force -Wait
```

Since `winget` itself requires the `Add-AppxProvisionedPackage` (which installs as a provisioned package + user), I realized there's no point to install the prerequisites as user using `Add-AppxPackage`, so might as well install everything with `Add-AppxProvisionedPackage`. Unless anyone can think of a good reason not to? This was initiated from #19. Details [here](https://github.com/asheroto/winget-install/issues/19#issuecomment-1890391852).

Incrementing to version 4.0.0.

### CODE REVIEW

**Comment by uffemcev** [comment]
Unfortunately, I get this error.
image

**Comment by uffemcev** [comment]
Manual installation also leads to an error.
image

**Comment by uffemcev** [comment]
Perhaps VClibs and Xaml are dependencies and need to be specified as a separate parameter. In this case, the installation is successful.

`Add-AppxProvisionedPackage -Online -PackagePath Microsoft.DesktopAppInstaller_8wekyb3d8bbwe.msixbundle -DependencyPackagePath Microsoft.UI.Xaml.2.7.x64.appx,Microsoft.VCLibs.x64.14.00.Desktop.appx -SkipLicense`

image

**Comment by asheroto** [comment]
Whoa, nice. Let me look at this closer and get back with you.

**Comment by asheroto** [comment]
Alright! I think I got it. Can you guys please try it now?

```powershell
&([ScriptBlock]::Create((irm https://raw.githubusercontent.com/asheroto/winget-install/v4/winget-install.ps1))) -Force -Wait
```

**Comment by uffemcev** [comment]
Looks like everything is working! 👌

```powershell
PS C:\Users\uffemcev> powershell "&([ScriptBlock]::Create((irm https://raw.githubusercontent.com/asheroto/winget-install/v4/winget-install.ps1))) -Force -Wait -Debug"
winget-install 4.0.0
To check for updates, run winget-install -CheckForUpdate

#################
# Prerequisites #
#################

Downloading VCLibs...
ОТЛАДКА: Downloading VCLibs from https://aka.ms/Microsoft.VCLibs.x64.14.00.Desktop.appx to
C:\Users\uffemcev\AppData\Local\Temp\tmpF36.tmp

Downloading UI.Xaml...
ОТЛАДКА: Downloading UI.Xaml from
https://github.com/microsoft/microsoft-ui-xaml/releases/download/v2.7.3/Microsoft.UI.Xaml.2.7.x64.appx to
C:\Users\uffemcev\AppData\Local\Temp\tmp14C5.tmp

##########
# winget #
##########

ОТЛАДКА: Getting latest release...
Downloading winget license...
ОТЛАДКА: Downloading winget license from https://github.com/microsoft/winget-cli/releases/download/v1.6.3482/24146eb205d040e69ef2d92d7034d97f_License1.xml to C:\Users\uffemcev\AppData\Local\Temp\tmp1C38.tmp

Downloading winget...
ОТЛАДКА: Downloading winget from https://aka.ms/getwinget to C:\Users\uffemcev\AppData\Local\Temp\tmp2717.tmp

Installing winget and its dependencies...

############
# Complete #
############

winget installed successfully.
Checking if winget is installed and working...
winget is installed and working now, you can go ahead and use it.

Waiting for 10 seconds before exiting...
```

**Review by uffemcev — APPROVED** [review]
👍

File: README.md [hunk 1]
```diff
@@ -150,6 +146,9 @@ As a more conventional approach, download the latest [winget-install.ps1](https:
 
 You can use the `-Force` or `-ForceClose` parameters if needed, or use `$Force = $true` and `$ForceClose = $true` global session variables if preferred.
 
+> [!TIP]
+> If for some reason your PowerShell window closes at the end of the script and you don't want it to, or don't want your other scripts to be interrupted, you can wrap the command in a `powershell -command "COMMAND HERE"`. For example, `powershell -command "irm asheroto.com/winget | iex"`.
```
Comment (uffemcev) [inline README.md:150]: `powershell "irm asheroto.com/winget | iex"` works too.

**Comment by asheroto** [comment]
Closed and manually merged. Thanks!

File: README.md (hunk 1 above)
Comment (asheroto) [reply README.md]: Good info! Changed it.

**Comment by asheroto** [comment]
Released and working. Thanks again guys!

**Pull request closed without merging**